test/tokenomics/rootGaugeLocal.ts

```diff
@@ -0,0 +1,238 @@
+
+const DAY = 86400
+const WEEK = DAY * 7
+const MAXTIME = DAY * 365 * 4
+const LOCK_START_TIMESTAMP = 2362003200
+
+const DUMMY_LP_TOKEN_NAME = "DUMMY_CHILD_CHAIN_LP_TOKEN"
+const REWARDS_ONLY_GAUGE_NAME = "RewardsOnlyGauge"
+const ROOT_GAUGE_LOCAL_NAME = "RootGaugeLocal"
+const GAUGE_CONTROLLER_NAME = "GaugeController"
+const CHILD_CHAIN_STRAMER_NAME = "ChildChainStreamer"
+
+describe("Root Gauge (Local)", () => {
+  let signers: Array<Signer>
+  let users: string[]
+  let deployer: Signer
+  let veSDL: VotingEscrow
+  let sdl: SDL
+  let minter: Minter
+  let gaugeController: GaugeController
+  let rootGauge: RootGaugeLocal
+
+  // side chain contracts
+  let childChainStreamer: ChildChainStreamer
+  let rewardsOnlyGauge: RewardsOnlyGauge
+  let dummyLpToken: GenericERC20
+  let rewardTokens: string[] = []
+
+  const setupTest = deployments.createFixture(
+    async ({ deployments, ethers }) => {
+      await deployments.fixture() // ensure you start from a fresh deployments
+
+      signers = await ethers.getSigners()
+      users = await Promise.all(
+        signers.map(async (signer) => signer.getAddress()),
+      )
+      deployer = signers[0]
+      minter = await ethers.getContract("Minter")
+
+      dummyLpToken = await ethers.getContract(DUMMY_LP_TOKEN_NAME)
+      rootGauge = await ethers.getContract(ROOT_GAUGE_LOCAL_NAME)
+      childChainStreamer = await ethers.getContract(CHILD_CHAIN_STRAMER_NAME)
+      rewardsOnlyGauge = await ethers.getContract(REWARDS_ONLY_GAUGE_NAME)
+      gaugeController = await ethers.getContract(GAUGE_CONTROLLER_NAME)
+      sdl = await ethers.getContract("SDL")
+      veSDL = await ethers.getContract("VotingEscrow")
+
+      // MAINNET TXs
+
+      if (await sdl.paused()) {
+        await sdl.enableTransfer()
+      }
+
+      await sdl.approve(veSDL.address, MAX_UINT256)
+
+      await veSDL.create_lock(
+        BIG_NUMBER_1E18.mul(10_000_000),
+        (await getCurrentBlockTimestamp()) + MAXTIME,
+      )
+
+      await sdl.transfer(minter.address, BIG_NUMBER_1E18.mul(10_000_000))
+
+      // SIDE CHAIN TXs
+      await dummyLpToken.mint(users[0], BIG_NUMBER_1E18)
+      await dummyLpToken.mint(users[10], BIG_NUMBER_1E18)
+      await dummyLpToken.approve(rewardsOnlyGauge.address, MAX_UINT256)
+      await dummyLpToken
+        .connect(signers[10])
+        .approve(rewardsOnlyGauge.address, MAX_UINT256)
+
+      rewardTokens = []
+      for (let i = 0; i < 10; i++) {
+        try {
+          rewardTokens.push(await rewardsOnlyGauge.reward_tokens(i))
+        } catch {
+          break
+        }
+      }
+    },
+  )
+
+  beforeEach(async () => {
+    await setupTest()
+
+    await setTimestamp(
+      Math.floor(((await getCurrentBlockTimestamp()) + WEEK) / WEEK) * WEEK,
+    )
+
+    // If rate is not initialized, initialize it
+    if ((await minter.rate()).eq(MAX_UINT256)) {
+      await minter.update_mining_parameters()
+    }
+    await rootGauge.connect(deployer).checkpoint()
```

Comment:
What's the rationale behind calling `checkpoint()` here?

Reply:
This was to mimic calling checkpoint every week. Since we need to advance the period twice to see the changes, need to call checkpoint twice as well.